auacm/app/static/js/controllers/EditCompetitionController.js

```diff
@@ -0,0 +1,223 @@
+app.controller('CreateCompetitionController', ['$scope', '$http', '$location',
+        '$routeParams', function($scope, $http, $location, $routeParams) {
+    $scope.compProblems = [];
+    var competition;
+
+    if ($routeParams.cid !== undefined) {
```

Comment:
I vote for `someVar === undefined`. To me, it's closer to the normal null check that would be done in any sensible language. As someone mentioned in the SO question, it's more concise than `typeof someVar === 'undefined'`. To me, it just makes more sense.